src/app/login/login.component.ts

```diff
@@ -1,10 +1,60 @@
 import { Component } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { AuthService } from '../services/auth.service';
+
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
   styleUrls: ['./login.component.scss']
 })
+
+
+
+
 export class LoginComponent {
+login: FormGroup<any>;
+form: any;
+
+
+emailFormControl = new FormControl('', [
+  Validators.required,
+  Validators.email,
+]);
+passwordFormControl = new FormControl('', [
+  Validators.required,
+]);
+  snackBar: any;
```

Comment:
these should be injected in constructor

Reply:
Not necessarily... this is because from what is explained, formControl validation method doesn't need you to declare the validator variables in the constructor

Reply:
But as @swoogie proposed, I will use the formbuilder approach